ipie/addons/adafqmc/qmc/adafqmc.py

```diff
@@ -0,0 +1,486 @@
+import numpy as np
+import torch
+from torch.utils.checkpoint import checkpoint
+from ipie.addons.adafqmc.hamiltonians.hamiltonian import ham_with_obs, rot_ham_with_orbs
+from ipie.addons.adafqmc.trial_wavefunction.sdtrial import SDTrial
+from ipie.addons.adafqmc.walkers.rhf_walkers import Walkers, initialize_walkers, reorthogonalize, sr
+from ipie.addons.adafqmc.propagation.propagator import Propagator
+import time
+
+
+class Dist_Params:
+    def __init__(
+        self,
+        num_walkers_per_process,
+        num_steps_per_block,
+        ad_block_size,
+        num_ad_blocks,
+        timestep,
+        stabilize_freq,
+        pop_control_freq,
+        pop_control_freq_eq,
+        seed,
+        grad_checkpointing=False,
+        chkpt_size=50,
+    ):
+        """
+        Parameters
+        -------
+        num_walkers_per_process : int
+            Number of walkers per process
+        num_steps_per_block : int
+            Number of steps per AD block
+        num_ad_blocks : int
+            Number of AD blocks
+
+        """
+        self.num_walkers_per_process = num_walkers_per_process
+        self.num_steps_per_block = num_steps_per_block
+        self.ad_block_size = ad_block_size
+        self.num_ad_blocks = num_ad_blocks
+        self.timestep = timestep
+        self.stabilize_freq = stabilize_freq
+        self.pop_control_freq = pop_control_freq
+        self.pop_control_freq_eq = pop_control_freq_eq
+        self.seed = seed
+        self.grad_checkpointing = grad_checkpointing
+        self.chkpt_size = chkpt_size
+        if self.grad_checkpointing and num_steps_per_block % chkpt_size != 0:
+            raise ValueError(
+                "The number of steps per block should be divisible by the checkpoint size"
+            )
+
+
+def prep_dist_adafqmc(
```

Comment:
prepare_distributed_adafqmc

Reply:
Why does this function exist? it looks like you're just passing the parameters to the class.

Reply:
Yes, probably put it in utils? This is just for convenience

Reply:
nah, utils is bad practice (I know we're terrible offenders for this). It's really a factory method for ADAFQMC, I would just add a build method similar to T = 0 AFQMC